core/src/main/scala/flow/Generator.scala

```diff
@@ -2,137 +2,329 @@ package datto.flow
 
 import akka.stream.{ ActorMaterializer, FlowShape, Graph }
 import akka.stream.scaladsl.{ Flow, Keep, RunnableGraph, Sink, Source }
+
 import scala.concurrent.{ ExecutionContext, Future }
 import scala.concurrent.duration._
 import scala.collection.immutable
 
+case class GeneratorOptions(onShutdown: () ⇒ Future[Unit]) {
+  def combine(other: GeneratorOptions)(implicit ec: ExecutionContext): GeneratorOptions =
+    GeneratorOptions { () ⇒
+      onShutdown().flatMap(_ ⇒ other.onShutdown())
+    }
+}
+
+object GeneratorOptions {
+  def fromFuture(futureOptions: Future[Option[GeneratorOptions]])(
+    implicit
```

Comment:
formatting seems off here, why is this on its own line

Reply:
Yeah it's so weird, normally scalafmt doesn't do this to me (it would split the implicit part onto its own line, but it wouldn't then also split up the implicit part into multiple lines!).  Could it be picking up different rules in the datto-flow project?

Anyway, as I mentioned in the standup, I'm going to try to put back _all_ of the extraneous reformatting, even the stuff that scalafmt normally does.  The diff is really hard to read, and I'd like it to be clear what is going on when we merged this.